src/aliceVision/lensCorrectionProfile/lcp.hpp

```diff
@@ -0,0 +1,733 @@
+
+    /**
+    * @brief Set an alernate model name for the lens
+    * @param[in] alternate model name
+    */
+    inline void addLensModel(std::string lensModel) { Lens.push_back(lensModel); }
+
+    /**
+    * @brief Set image width
+    * @param[in] image width
+    */
+    inline void setImageWidth(int w) { ImageWidth = w; }
+
+    /**
+    * @brief Set image length
+    * @param[in] image length
+    */
+    inline void setImageLength(int l) { ImageLength = l; }
+
+    /**
+    * @brief Set sensor format factor
+    * @param[in] sensor format factor
+    */
+    inline void setSensorFormatFactor(float f) { SensorFormatFactor = f; }
+
+    /**
+    * @brief Set raw profile status
+    * @param[in] raw profile status
+    */
+    inline void setAsRawProfile() { CameraRawProfile = true; }
+
+private:
+    // XML handlers
+    static void XmlStartHandler(void* pLCPinfo, const char* el, const char** attr);
+    static void XmlEndHandler(void* pLCPinfo, const char* el);
+    static void XmlStartHandlerCommonOnly(void* pLCPinfo, const char* el, const char** attr);
+    static void XmlEndHandlerCommonOnly(void* pLCPinfo, const char* el);
+    static void XmlTextHandler(void* pLCPinfo, const char* s, int len);
+
+    // Loading states control
+    inline bool isSeqOpened() { return _isSeqOpened; }
+    inline void openSequence() { _isSeqOpened = true; }
+    inline void closeSequence() { _isSeqOpened = false; }
+
+    inline bool isCommonOK() { return _isCommonOK; }
+    inline void setCommonOK() { _isCommonOK = true; }
+    inline void unsetCommonOK() { _isCommonOK = false; }
+
+    inline bool isCamDataOK() { return _isCamDataOK; }
+    inline void setCamDataOK() { _isCamDataOK = true; }
+    inline void unsetCamDataOK() { _isCamDataOK = false; }
+
+    inline bool isAlternateLensIDsOpened() { return _inAlternateLensIDs; }
+    inline void openAlternateLensIDs() { _inAlternateLensIDs = true; }
+    inline void closeAlternateLensIDs() { _inAlternateLensIDs = false; }
+
+    inline bool isAlternateLensNamesOpened() { return _inAlternateLensNames; }
+    inline void openAlternateLensNames() { _inAlternateLensNames = true; }
+    inline void closeAlternateLensNames() { _inAlternateLensNames = false; }
+
+    inline bool isWaitPerspModeldescription() { return _waitPerspModeldescription; }
+    inline void setWaitPerspModeldescription() { _waitPerspModeldescription = true; }
+    inline void unsetWaitPerspModeldescription() { _waitPerspModeldescription = false; }
+
+    inline bool isGetText() { return _getText; }
+    inline void setGetText() { _getText = true; }
+    inline void unsetGetText() { _getText = false; }
+
+    inline int getModelCount() { return _modelCount; }
+    inline void increaseModelCount() { _modelCount++; }
+
+    inline void storeCurrParams() { v_lensParams.push_back(currLensParam); }
+
+    LensParam currLensParam;
+
+    bool search(settingsInfo& settings, LCPCorrectionMode mode, int& iLow, int& iHigh, float& weightLow);
```

Comment:
Seems like a quite important method, could we get some documentation ? (same for `combine`)